compiler/rustc_session/src/config.rs

```diff
@@ -2497,6 +2505,10 @@ pub fn build_session_options(early_dcx: &mut EarlyDiagCtxt, matches: &getopts::M
     let mut unstable_opts = UnstableOptions::build(early_dcx, matches, &mut target_modifiers);
     let (lint_opts, describe_lints, lint_cap) = get_cmd_lint_options(early_dcx, matches);
 
+    if !unstable_opts.unstable_options && json_section_timings {
```

Comment:
Huh, weird that we don't have a single place to check when `-Zunstable-options` is necessary. Which means that every error message of this kind is slightly different.

Reply:
Yeah, CLI handling is quite arbitrary here.